tests/models/test_horovod.py

```diff
@@ -235,6 +241,113 @@ def get_optimizer_params(optimizer):
     assert get_model_params(model.generator) == get_optimizer_params(trainer.optimizers[0])
     assert get_model_params(model.discriminator) == get_optimizer_params(trainer.optimizers[1])
 
+
+@pytest.mark.skipif(not HOROVOD_AVAILABLE, reason="Horovod is unavailable")
+@pytest.mark.skipif(platform.system() == "Windows", reason="Horovod is not supported on Windows")
+def test_result_reduce_horovod(tmpdir):
+    """Make sure result logging works with Horovod."""
+    tutils.reset_seed()
+    tutils.set_random_master_port()
+
+    def hvd_test_fn():
+        path_here = os.path.abspath(os.path.dirname(__file__))
+        path_root = os.path.abspath(os.path.join(path_here, '..', '..'))
+        sys.path.insert(0, os.path.abspath(path_root))
+
+        from tests.base.boring_model import BoringModel
+
+        import horovod.torch as hvd
+
+        class TestModel(BoringModel):
+            def training_step(self, batch, batch_idx):
+                self.training_step_called = True
+
+                tensor = torch.tensor([1.0])
+                self.log("test_tensor", tensor, sync_dist=True, sync_dist_op='sum',
+                         on_step=True, on_epoch=True)
+
+                res = self._results
+                assert res["test_tensor"].item() == hvd.size(), \
```

Comment:
Not sure I understand this test. Could you provide an explanation ?

Reply:
Yes, this test mirrors the one in `tests/core/test_results.py::_ddp_test_fn`.  The purpose is to check that `tensor` (which starts as `1.0`) is summed across ranks so its final value is `hvd.size()` (the number of workers).

Reply:
Thanks !